db/memtable_list.cc

```diff
@@ -430,6 +465,8 @@ Status MemTableList::TryInstallMemtableFlushResults(
                            cfd->GetName().c_str(), m->file_number_, mem_id);
           assert(m->file_number_ > 0);
           current_->Remove(m, to_delete);
+          UpdateMemoryUsageExcludingLast();
+          ResetTrimHistoryNeeded();
```

Comment:
See my comments in `InstallMemtableAtomicFlushResults()`.

Reply:
Chatted offline. The previous `current_->Remove` will trigger possible trimming of `memtable_history_`. Therefore, in either case we can reset `trim_history_needed_` flag.